### DISCUSSION TITLE
Ability to add a new upload within Rich Text

### DISCUSSION THREAD

**Discussion by jmikrut** [payloadcms/payload#1233] [Roadmap]
Right now, the Rich Text Upload element allows you to select from existing uploads. In addition, it should allow you to add a new upload as well.

As asked for by `asheboy` on Discord. Marking as planned!

**Comment by Sccttt** [comment]
Hey @jmikrut,

Ive submitted a PR which should achieve this. https://github.com/payloadcms/payload/pull/1336

**Reply by jmikrut** [reply]
Hey @Sccttt — I left some feedback regarding your PR.

The team is discussing this at the moment and we have some ideas for how to re-purpose our "nested editing" functionality that we recently released, which could be re-used here. We want to leverage that new pattern and are going to commit to building this shortly.

Related discussion:
https://github.com/payloadcms/payload/discussions/763
